packages/performance/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/performance/FirebasePerformancePlugin.java

```diff
@@ -116,4 +123,180 @@ public void isEnabled(PluginCall call) {
+            if (attribute == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_ATTRIBUTE_MISSING);
+                return;
+            }
+            Trace trace = implementation.getTraceByName(traceName);
+            if (trace == null) {
+                call.reject(ERROR_TRACE_NOT_FOUND);
+                return;
+            }
+            String value = FirebasePerformance.getAttribute(trace, attribute);
+            JSObject result = new JSObject();
+            result.put("value", value);
+            call.resolve(result);
+        } catch (Exception exception) {
+            Logger.error(TAG, exception.getMessage(), exception);
+            call.reject(exception.getMessage());
+        }
+    }
+
+    @PluginMethod(returnType = PluginMethod.RETURN_PROMISE)
+    public void getAttributes(PluginCall call) {
+        try {
+            String traceName = call.getString("traceName");
+            if (traceName == null) {
+                call.reject(ERROR_TRACE_NAME_MISSING);
+                return;
+            }
+            Trace trace = implementation.getTraceByName(traceName);
+            if (trace == null) {
+                call.reject(ERROR_TRACE_NOT_FOUND);
+                return;
+            }
+            Map<String, String> attributesMap = FirebasePerformance.getAttributes(trace);
+            JSObject result = new JSObject();
+            JSObject resultMap = new JSObject();
+            for (String attribute : attributesMap.keySet()) {
+                resultMap.put(attribute, attributesMap.get(attribute));
+            }
+            result.put("result", resultMap);
+            call.resolve(result);
+        } catch (Exception exception) {
+            Logger.error(TAG, exception.getMessage(), exception);
+            call.reject(exception.getMessage());
+        }
+    }
+
+    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
+    public void removeAttribute(PluginCall call) {
+        try {
+            String traceName = call.getString("traceName");
+            if (traceName == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_TRACE_NAME_MISSING);
+                return;
+            }
+            String attribute = call.getString("attribute");
+            if (attribute == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_ATTRIBUTE_MISSING);
+                return;
+            }
+            Trace trace = implementation.getTraceByName(traceName);
+            if (trace == null) {
+                call.reject(ERROR_TRACE_NOT_FOUND);
+                return;
+            }
+            FirebasePerformance.removeAttribute(trace, attribute);
+            call.resolve();
+        } catch (Exception exception) {
+            Logger.error(TAG, exception.getMessage(), exception);
+            call.reject(exception.getMessage());
+        }
+    }
+
+    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
```

Comment:
```suggestion
    @PluginMethod()
```

`RETURN_NONE` means that the return type of the TS function is `void`. No promise is returned in that case. However, it should be `Promise<void>`. Please fix this for alle methods. The Capacitor docs are incorrect here, see [this comment](https://forum.ionicframework.com/t/writing-a-capacitor-pluginmethod-for-android-annotated-with-returntype-pluginmethod-return-none/221721/4?u=robingenz).